src/api/style.js

```diff
@@ -185,7 +187,7 @@ export default class Style {
             this._styleSpec.strokeColor,
             this._styleSpec.strokeWidth,
             this._styleSpec.filter,
-        ].filter(x => x && x._getMinimumNeededSchema);
+        ].concat(Object.keys(this._styleSpec.variables).map(varname => this._styleSpec.variables[varname])).filter(x => x && x._getMinimumNeededSchema);
```

Comment:
IMO this would be easier to read and understand
```js
.concat(this._getVariables())
.filter(x => x && x._getMinimumNeededSchema);
```


```js
_getVariables() {
  return Object.values(this._styleSpec.variables);
}
```